web/packages/teleterm/src/mainProcess/agentRunner.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { spawn, ChildProcess } from 'node:child_process';
+import os from 'node:os';
+
+import Logger from 'teleterm/logger';
+import { RootClusterUri } from 'teleterm/ui/uri';
+
+import { generateAgentConfigPaths } from './createAgentConfigFile';
+import { AgentProcessState, RuntimeSettings } from './types';
+
+const MAX_STDERR_LINES = 10;
+
+export class AgentRunner {
+  private logger = new Logger('AgentRunner');
+  private agentProcesses = new Map<RootClusterUri, ChildProcess>();
+
+  constructor(
+    private settings: RuntimeSettings,
+    private sendProcessState: (
+      rootClusterUri: RootClusterUri,
+      state: AgentProcessState
+    ) => void
+  ) {}
+
+  /**
+   * Starts a new agent process.
+   * If an existing process exists for the given root cluster, the old one will be killed with SIGKILL.
+   * To kill the old process gracefully before starting the new one, use `kill()`.
+   */
+  start(rootClusterUri: RootClusterUri): void {
+    if (this.agentProcesses.has(rootClusterUri)) {
+      this.agentProcesses.get(rootClusterUri).kill('SIGKILL');
+      this.logger.warn(`Forcefully killed agent process for ${rootClusterUri}`);
+    }
```

Comment:
I think it can happen during the setup. If for some reason, the node won't be available in the cluster within the specified timeout, we will show "Retry" button. When the user clicks it, we will call `start()` again. If we won't close the previous agent, the user may end up with two instances of it.

Btw, I replaced SIGKILL with the new `killProcess` function.